src/query/atomWithInfiniteQuery.ts

```diff
@@ -88,6 +89,15 @@ export function atomWithInfiniteQuery<
           | { data?: undefined; error: TError }
       ) => {
         if (result.error) {
+          if (isCancelledError(result.error)) {
+            // fetch has been canceled,
+            // revert to prevData
+            if (settlePromise) {
+              if (prevData) settlePromise(prevData)
+              settlePromise = null
```

Comment:
```suggestion
              if (prevData) {
                settlePromise(prevData)
                settlePromise = null
              }
```
settlePromise should be cleared only if it's invoked.

Reply:
resolving with prevData is required for refetch? hm but this file is atomWithInfiniteQuery, not atomWithQuery.
I feel this a bit weird because settlePromise and setData are always paired.

Can we do
```js
        if (result.error && !isCancelledError(result.error)) {
```
in L91?

Reply:
Yep that works too, wasn't sure if we needed to do the settledPromise stuff (tbh not clear on what it does exactly lol)

Reply:
I know. This is because of very tricky suspense behavior. I'm not even 100% sure, so adding tests to check the behavior would be important.